### PR TITLE
[Android] Fix Keyboard.Numeric issue

### LINKED ISSUE #11865: DisplayPromptAsync : Keyboard.Numeric broken in release mode
### Description

DisplayPromptAsync doesn't work at all in release mode when using numeric keyboard.
- Works in DEBUG : when tapping on virtual keyboard, the inputs is set : works well
- Doesn't work in RELEASE : when tapping on virtual keyboard, nothing happens

What is weird is that the Telephone keyboard type works.

See github repo to reproduce.

### Steps to Reproduce

Don't forget to set the configuration to RELEASE !

image
### Link to public reproduction project repository

https://github.com/nk54/maui_bug_prompt

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11 and up

### Did you find any workaround?

I'm using Telephone keyboard instead until it is fixed

### Relevant log output

_No response_

### LINKED ISSUE #15884: [regression/8.0-preview5] Entry not working when keyboard is numeric and os language is not english
### Description

On Android, using net8-pre5, Entry control is not working when Keyboard is Numeric and OS language is not english.
Both in Debug and Release mode.
This was working in net7.

### Steps to Reproduce

1. Create a Maui app for Android using net8-pre5
2. In XAML, add an Entry and set Keyboard to Numeric
3. Change OS language to any non-english language (for example, French)
4. Run app, tap on the entry, it's impossible to enter a value.

### Link to public reproduction project repository

https://github.com/tranb3r/Issues/tree/main/MauiAppKeyboardNumeric

### Version with bug

8.0.0-preview.5.8529

### Last version that worked well

7.0.86

### Affected platforms

Android

### Affected platform versions

Android 33

### Did you find any workaround?

No workaround

### Relevant log output

_No response_

### LINKED ISSUE #16429: Entry Control does not accepting numeric inputs when Android language set to Arabic
### Description

Issue with Numric keyboard and Arabic language only 

### Steps to Reproduce

Create new project
On default maui project add Entry Control

Run simulator or test it on a real device 

Switch phone language to Arabic 

While trying to input any numric, the control will not resopnsive

`               <Entry  Keyboard="Numeric"  />
`

### Link to public reproduction project repository

N/A

### Version with bug

8.0.0-preview.5.8529

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 9 and up

### Did you find any workaround?

Use Telephone type insted of Numeric

`Keyboard="Telephone"`

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix Keyboard.Numeric issue on Android.

fix-keyboard-alert

Change the Android device or simulator language to French and launch the .NET MAUI Gallery. Navigate to Controls > Entry. Try to introduce decimals using the comma as separator in the entry with Keyboard.Numeric

### Issues Fixed

Fixes #11865
Fixes #15884
Fixes #16429

### CODE REVIEW

File: src/Core/src/Platform/Android/LocalizedDigitsKeyListener.cs [hunk 1]
```diff
@@ -215,7 +215,7 @@ bool IsDecimalPointChar(char c)
 				}
 			}
 
-			return stripped ?? filterFormatted ?? new String("");
+			return stripped ?? filterFormatted;
```
Comment (rachelkang) [inline src/Core/src/Platform/Android/LocalizedDigitsKeyListener.cs:218]: This appears to fix the issue, but can you help me understand why? Why wasn't it working before with other system display languages and why does it work now with this change?

**Review by rachelkang — CHANGES_REQUESTED** [review]
This issue and fix are not tied to display prompt at all, so I suggest removing the sample from the Alerts Page and adding a simple `<Entry Keyboard="Numeric" Text="Numeric text" />` to the EntryPage.xaml instead

(The original issue used a numeric keyboard entry both within and without a display prompt)

other than that, LGTM!

**Comment by jsuarezruiz** [comment]
> This issue and fix are not tied to display prompt at all, so I suggest removing the sample from the Alerts Page and adding a simple `<Entry Keyboard="Numeric" Text="Numeric text" />` to the EntryPage.xaml instead
> 
> (The original issue used a numeric keyboard entry both within and without a display prompt)
> 
> other than that, LGTM!

Right, we already have that sample in the Entry page https://github.com/dotnet/maui/blob/main/src/Controls/samples/Controls.Sample/Pages/Controls/EntryPage.xaml#L207

**Review by rachelkang — CHANGES_REQUESTED** [review]
LGTM! I just recommend removing the sample as per my last comment https://github.com/dotnet/maui/pull/16163#issuecomment-1679186446

**Review by rachelkang — APPROVED** [review]

File: src/Core/src/Platform/Android/LocalizedDigitsKeyListener.cs (hunk 1 above)
Comment (mattleibow) [reply src/Core/src/Platform/Android/LocalizedDigitsKeyListener.cs:218]: Yeah, Why??? 😭

**Review by mattleibow — APPROVED** [review]

**Pull request merged**